x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/hooks/use_can_access_some_artifacts.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { useEndpointPrivileges } from '../../../../../../common/components/user_privileges/endpoint';
+
+/**
+ * Checks to see if the current user can access at least one artifact page.
+ * Note that this hook will return `false` if the Authz is still being loaded.
+ */
+export const useCanAccessSomeArtifacts = (): boolean => {
```

Comment:
Would be better returning also the `loading` state? So when we use this hook we know if it is loading or not before relying on his response

Reply:
Not sure. The check to see if authz is still loading is happening in the component that uses this hook by using the value directly from the `EndpointAuthz`, so loading is being displayed.  Also, this hook is not intended to check Authz, only to inform if the user can access/see at least one artifact type.

I'll leave it as is, unless you feel strongly about it.